crates/hyperswitch_domain_models/src/payment_method_data.rs

```diff
@@ -289,10 +289,20 @@ pub struct CryptoData {
 
 #[derive(Debug, Clone, Eq, PartialEq, serde::Deserialize, serde::Serialize)]
 #[serde(rename_all = "snake_case")]
-pub struct UpiData {
+pub enum UpiData {
+    UpiCollect(UpiCollectData),
+    UpiIntent(UpiIntentData),
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(rename_all = "snake_case")]
+pub struct UpiCollectData {
```

Comment:
we have changed a struct to enum, is this change backwards compable? If this breaks existing api contract then this change cannot be made.

Reply:
Through stripe compatibility layer, backwards compatibility is handled.